lightningd/plugin.c

```diff
@@ -984,9 +984,8 @@ static void plugin_config(struct plugin *plugin)
 	list_for_each(&plugin->plugin_opts, opt, list) {
 		/* Trim the `--` that we added before */
 		name = opt->name + 2;
-		if (!opt->value)
-			opt->value = "";
-		json_add_string(req->stream, name, opt->value);
+		if (opt->value)
```

Comment:
Could this be because there is some difference between the existence of `--option` vs. its nonexistence?

That is, could it be that `--option` implicitly means `--option=""` and the non-existence of `--option` means it is not placed?

@cdecker seems to be related to nearby code, so maybe should check.

Reply:
No, this is called when --option isn't specified at all, so that can't be it.